src/main/java/core/gui/comp/entry/ColorLabelEntry.java

```diff
@@ -236,15 +236,39 @@ private void setGraphicalChangeValue(double number) {
                 true);
     }
 
+    private void initAdd() {
+        var isInitDone = this.getLayout() != null && this.getLayout().getClass() == BoxLayout.class;
+        if (!isInitDone){
+            this.setLayout(new BoxLayout(this, BoxLayout.X_AXIS));
+            var icon = this.getIcon();
+            if (icon!=null){
+                super.setIcon(null);
+                this.add(new JLabel(icon));
```

Comment:
Doesn't this get called one time too many the first time in `addIcon`?  I must admit I don't follow the logic. :)

Reply:
this should handle the case if setIcon was called before addIcon is called. This icon will be added to the list of components. The added icon is then appended. (that's the plan)